cfme/common/__init__.py

```diff
@@ -85,6 +86,75 @@ def _assign_unassign_policy_profiles(self, assign, *policy_profile_names):
 
+    def assign_policy_profiles_multiple_entities(self, entities, *policy_profile_names):
+        """ Assign Policy Profiles to selected entity's on Collection All view
+
+        Args:
+            entities: list of entity's from collection table
+            policy_profile_names: :py:class:`str` with Policy Profile names. After Control/Explorer
+                coverage goes in, PolicyProfile objects will be also passable.
+        """
+        map(self.assigned_policy_profiles.add, policy_profile_names)
+        self._assign_or_unassign_policy_profiles_multiple_entities(
+            entities, True, *policy_profile_names)
+
+    def unassign_policy_profiles_multiple_entities(self, entities, *policy_profile_names):
+        """ UnAssign Policy Profiles to selected entity's on Collection All view
+
+        Args:
+            entities: list of entity's from collection table
+            policy_profile_names: :py:class:`str` with Policy Profile names. After Control/Explorer
+                coverage goes in, PolicyProfile objects will be also passable.
+        """
+        for pp_name in policy_profile_names:
+            try:
+                self.assigned_policy_profiles.remove(pp_name)
+            except KeyError:
+                pass
+        self._assign_or_unassign_policy_profiles_multiple_entities(
+            entities, False, *policy_profile_names)
+
+    def _assign_or_unassign_policy_profiles_multiple_entities(
+            self, entities, assign, *policy_profile_names):
+
+        """DRY function for managing policy profiles.
+
+        See :py:func:`assign_policy_profiles_multiple_entities`
+         and :py:func:`unassign_policy_profiles_multiple_entities`
+
+        Args:
+            entities: list of entity's from collection table
+            assign: Wheter to assign or unassign.
+            policy_profile_names: :py:class:`str` with Policy Profile names.
+        """
+        view = navigate_to(self, 'All')
+
+        # set item per page for maximum value in order to avoid paging,
+        # that will cancel the already check entity's
+        view.paginator.set_items_per_page(1000)
+
+        # check the entity's on collection ALL view
+        for entity in entities:
+            view.entities.get_entity(name=entity.name, id=entity.id).check()
+
+        wait_for(lambda: view.toolbar.policy.is_enabled, num_sec=5,
+                 message='Policy drop down menu is disabled after checking some entities')
+        view.toolbar.policy.item_select('Manage Policies')
+        # get the object of the Manage Policies view
+        manage_policies_view = self.create_view(navigator.get_class(self, 'ManagePolicies').VIEW)
+
+        for policy_profile in policy_profile_names:
+            if assign:
+                manage_policies_view.policy_profiles.check_node(policy_profile)
+            else:
+                manage_policies_view.policy_profiles.uncheck_node(policy_profile)
+        if manage_policies_view.save.disabled:
```

Comment:
**Required**

This at least needs to be logged - there's no assertion that save is disabled because there was no change to the assigned policy profiles.  This could easily hide errors, or return cleanly after not actually assigning/unassigning profiles.

A small enhancement to `wt.patternfly.CheckableBoostrapTree.fill()` would allow for its use here as well, instead of calling check_node/uncheck_node directly and return a 'changed' boolean, but that's not really in the scope here.

Reply:
I wanted to use this wt.patternfly.CheckableBoostrapTree.fill(),
From my understanding this method only can be use instead of calling check_node.
I don't see any unfill that can be use instead of uncheck_node!
should i ask Milan to add this simple unfill method?

Reply:
https://github.com/RedHatQE/widgetastic.patternfly/issues/42

I wrote this issue for it, because the whole time I was reading your PR I was scouring CheckableBootstrapTreeview for a cleaner way to call this, and it just doesn't support it.

Reply:
Waiting for enhance of the fill method, please let me know when its done.

Reply:
Use wt.patternfly.CheckableBoostrapTree.fill() instead of calling check_node/uncheck_node directly.